tests/ui-fulldeps/stable-mir/check_item_kind.rs

```diff
@@ -0,0 +1,84 @@
+// run-pass
+//! Test that item kind works as expected.
+
+// ignore-stage1
+// ignore-cross-compile
+// ignore-remote
+// ignore-windows-gnu mingw has troubles with linking https://github.com/rust-lang/rust/pull/116837
+// edition: 2021
+
+#![feature(rustc_private)]
+#![feature(assert_matches)]
+#![feature(control_flow_enum)]
+
+extern crate rustc_middle;
+#[macro_use]
+extern crate rustc_smir;
+extern crate rustc_driver;
+extern crate rustc_interface;
+extern crate stable_mir;
+
+use rustc_middle::ty::TyCtxt;
+use rustc_smir::rustc_internal;
+use stable_mir::*;
+use std::io::Write;
+use std::ops::ControlFlow;
+
+const CRATE_NAME: &str = "input";
+
+/// This function uses the Stable MIR APIs to get information about the test crate.
+fn test_item_kind(_tcx: TyCtxt<'_>) -> ControlFlow<()> {
+    let items = stable_mir::all_local_items();
+    assert_eq!(items.len(), 4);
+    // Constructor item.
+    for item in items {
+        let expected_kind = match item.name().as_str() {
+            "Dummy" => ItemKind::Ctor(CtorKind::Fn),
+            "dummy" => ItemKind::Fn,
+            "unit" => ItemKind::Fn,
```

Comment:
```suggestion
            "DummyUnit" => ItemKind::Fn,
```

right?

Reply:
wait I misread this, I mean:

```suggestion
            "unit" => ItemKind::Fn,
            "DummyUnit" => ItemKind::Ctor(CtorKind::Const),
```

Reply:
I don't think DummyUnit has a body, so it won't be listed as a CrateItem today. I.e.: The CtorKind(Const) is probably unreachable.

That said, I am not 100% confident that's the case. So I am OK with using the solution here, since it's better than crashing.

Reply:
Well in that case, let's just see if the test passes! 😆

Reply:
I thought that's what you were trying to say here:

> Actually, sorry for the back and forth. I looked a bit deeper -- I don't think that unit structs have a body at all.
> 
> Maybe let's leave this `unreachable!()`, or something...? What do you think?

Reply:
Kind of? Does `all_local_items` only return things with bodies?

Reply:
Yes. `CrateItem` is something that we need to polish a bit, see https://github.com/rust-lang/project-stable-mir/issues/34 for the discussion.

`all_local_items` returns everything that `mir_keys(())` query does.

Reply:
The other way to get a `CrateItem` is by converting a `StaticDef` or an `Instance` of type Item that has a body.

Reply:
👍 

cool, so technically we'll never get `CtorKind::Const` here anyways... but let's just keep it

Reply:
We do have an assumption today that a `CrateItem` has a body, since the `body()` call crashes if it doesn't.